File: tests/fixtures/java/classes.java
package com.example.models;

import java.util.List;
import java.util.Map;
import java.io.Serializable;

/// Simple model class with fields and methods.
public class Job implements Serializable {
    private String name;
    public int priority;
    protected List<String> tags;

    public Job(String name, int priority) {
        this.name = name;
        this.priority = priority;
    }

    public void reset() {
        name = "";
    }

    private static int computeHash(String input) {
        return input.hashCode();
    }
}

class InternalHelper {
    public String format(String value) {
        return "[" + value + "]";
    }
}

public class GenericContainer<T extends Comparable<T>> {
    private T value;
    public int count;

    public void add(T item) {
        count++;
    }

    public <U> U transform(T input) {
        return null;
    }
}

File: tests/fixtures/csharp/classes.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace TestProject.Models
{
    /// <summary>Simple model class with properties.</summary>
    [DataContract]
    public class Job
    {
        [Required]
        [DataMember(Name = "id")]
        public Guid Id { get; set; }

        [Required]
        public string Name { get; set; }

        private readonly int _priority;
        public int Priority => _priority;

        public static readonly List<string> ValidStates = new();

        public Job(string name, int priority)
        {
            Name = name;
            _priority = priority;
        }

        public void Reset()
        {
            Name = string.Empty;
        }

        private static int ComputeHash(string input) => input.GetHashCode();
    }

    /// <summary>Internal helper class.</summary>
    internal class InternalHelper
    {
        public string Format(string value) => $"[{value}]";
    }

    public class GenericContainer<T> where T : class
    {
        public T Value { get; set; }
        public int Count { get; private set; }

        public void Add(T item) { Count++; }
    }
}
